### PR TITLE
Fix gizmos transforms for components in bones

### PR DESCRIPTION
**Pull request by keianhzo**
This PR fixes gizmo transform issues when a component is attached to a bone. 

Related: https://github.com/MozillaReality/hubs-blender-exporter/pull/68#issuecomment-1168532385

### CODE REVIEW

**Comment by keianhzo** [comment]
@Exairnous In case you want to take a look

**Comment by Exairnous** [comment]
Yes, thank you.   I'm looking into it now.

**Comment by Exairnous** [comment]
I'm not sure if this is just an incremental fix to address bone gizmos not respecting object transforms or if it's for all the bone gizmo transform issues, but I decided to assume it was for the latter and have left some comments and code for that.  Updating bone gizmos after duplication is still needed, but that can be another PR.

Here is a screenshot to show that the code I have left matches Hubs:
hubs_blender_addon_gizmo_fixed

File: addons/io_hubs_addon/components/definitions/audio_zone.py [hunk 1]
```diff
@@ -30,18 +30,20 @@ class AudioZone(HubsComponent):
...
-            _, rot, _ = mat.decompose()
-            loc = bone.tail
+            mat_offset = Matrix.Translation(bone.tail - bone.head)
+            mat = ob.matrix_world @ mat_offset @ Matrix.Translation(
+                loc) @ rot.normalized().to_matrix().to_4x4() @ scaleMat
         else:
             loc, rot, scale = ob.matrix_world.decompose()
             scaleMat = Matrix.Diagonal(scale).to_4x4()
+            mat = Matrix.Translation(
+                loc) @ rot.normalized().to_matrix().to_4x4() @ scaleMat
```
[the remark this answers is not in the document]
Comment (keianhzo) [reply addons/io_hubs_addon/components/definitions/audio_zone.py:43]: Copy paste mistake. Updated.

File: addons/io_hubs_addon/components/definitions/audio_zone.py [hunk 2]
```diff
@@ -30,18 +30,20 @@ class AudioZone(HubsComponent):
...
             _, _, wScale = ob.matrix_world.decompose()
+            loc, rot, _ = bone.matrix.to_4x4().decompose()
             scaleMat = Matrix.Diagonal(bone.bbone_scaleout).to_4x4(
             ) @ Matrix.Diagonal(wScale).to_4x4()
-            mat = ob.matrix_world @ bone.matrix.to_4x4()
-            _, rot, _ = mat.decompose()
-            loc = bone.tail
+            mat_offset = Matrix.Translation(bone.tail - bone.head)
+            mat = ob.matrix_world @ mat_offset @ Matrix.Translation(
+                loc) @ rot.normalized().to_matrix().to_4x4() @ scaleMat
```
[the remark this answers is not in the document]
Comment (Exairnous) [reply addons/io_hubs_addon/components/definitions/audio_zone.py]: Interesting.  You're right that parented objects connect to the tail, but if you enter pose mode and transform bones, the bone origin is definitely the head.  Also, if you snap a bone to the 3D cursor, it snaps to the head, and vice versa.  Having a waypoint start from the tail would make it more difficult to place on the ground, for example (the other components would face similar issues), so I'm thinking that keeping gizmos (and Hubs) centered on the head is probably better, but it is an interesting discrepancy.

**Pull request merged**